### ISSUE TITLE
Provide possibility to rotate db certificates w/o ZITADEL restart

### ISSUE THREAD

**Issue by livio-a** [zitadel/zitadel#8594]
With the current pgx db connection pool, the certificates for connecting to the database are only read once (on startup).
Clients rotating their certificates on the database (on a regular basis) need to restart ZITADEL in order to reconnect to the database.

We should provide the possibility to exchange the certificate and reconnect to the database after the connection fails, resp. always read the cert from path.

### Acceptance criteria
- [ ] The connection pool automatically uses the new certificate if it got exchanged

**Comment by eliobischof** [comment]
I think we need a solution that resolves this together with https://github.com/zitadel/zitadel/issues/7962.

I like the idea of having file watchers on configured paths because it wouldn't require any additional interaction with ZITADEL apart from updating the files.

**Comment by kfcss** [comment]
Notice that it would be nice to use the new certificates before a connection fails. 
It would be nice we could provide a DB connection max lifetime to ensure connections are rotated onto new certificates.

**Comment by livio-a** [comment]
> Notice that it would be nice to use the new certificates before a connection fails. It would be nice we could provide a DB connection max lifetime to ensure connections are rotated onto new certificates.

From what I know, this issue arose with the exchange of the db driver. With the old it used to worked, so I assume you exchanged the (mounted) certificates on the same path, correct?
So having a file watcher / SIGHUP or even getting the current certificate without caching it, would again solve the issue.

**Comment by kfcss** [comment]
> From what I know, this issue arose with the exchange of the db driver. With the old it used to worked, so I assume you exchanged the (mounted) certificates on the same path, correct? So having a file watcher / SIGHUP or even getting the current certificate without caching it, would again solve the issue.

Yes, we changed the certificates on-disk. What we actually have is a Kubernetes Pod that mounts a secret to a path. The files on disk will be updated by Kubernetes when the secret is updated.

I am pretty sure that the old driver had a connection max-lifetime set. It thus recreated SQL connections once in a while and it always used certificates freshly read from disk.

So a different solution is to re-create connections at some interval, using certificates on disk when new connections are made.

**Comment by linwalth** [comment]
AFAIK, CNPG is the recommended database backend for self-hosted installations.
CNPG recreates its server certs every 90 days and Zitadel starts running into errors whenever the CNPG server certs used as db certs are updated.
Thus it would be somewhat vital to have this issue resolved :)

**Comment by linwalth** [comment]
Still relevant to our interests.
